Realm/Realm/Realm.cs

```diff
@@ -1364,6 +1359,46 @@ internal void ExecuteOutsideTransaction(Action action)
 
         #endregion Transactions
 
+        internal class RealmMetadata
+        {
+            private readonly Dictionary<string, RealmObjectBase.Metadata> stringToRealmObjectMetadataDict;
+            private readonly Dictionary<TableKey, RealmObjectBase.Metadata> tableKeyToRealmObjectMetadataDict;
+
+            public IEnumerable<RealmObjectBase.Metadata> Values => stringToRealmObjectMetadataDict.Values;
```

Comment:
TODO: check where/why this is used.

Reply:
It is used in `realm.RemoveAll()`. It makes sense that it is used there, but probably would be more appropriate to transform the property in a method called `GetAllRealmObjectsMetadata()` or similar

Reply:
It's probably best to just implement `RemoveAll` natively. It's obviously not a hot path, but it may have peculiar side effects - e.g. if you open the Realm file with a subset of its schema, you likely expect `RemoveAll` to leave the file empty, but it'll remove only some data.

You can either leave it as is and open an issue to refactor `RemoveAll` or go ahead and fix it in this PR 😛

Reply:
I have added a new issue (#2233). Too much stuff already in this PR :)